apis/logging/v1/input_source_types.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NOTE: The Enum validation on SourceSpec.Type must be updated if the list of types changes.
+
+// SourceSpec is a union of input source types.
+//
+// The fields of this struct define the set of known source types.
+type SourceSpec struct {
+	HTTP *HTTPSource `json:"http,omitempty"`
+}
+
+const (
+	FormatK8S  = "k8s"  // Log events in k8s list format, e.g. API audit log events.
```

Comment:
maybe we should be more specific.  Isn't this actually a serialized kubernetes event as yaml?  If so then we should name it something like kubernetesEvent or the like

Reply:
Yes, this needs more thought. I was thinking of a generic k8s "list" format, using the standard k8s resource list YAML. The type of object in the list would be indicated by the standard meta GVK fields.
We could be more specific "Event.audit.k8s.io/v1` , but I'm not sure how specific we want to be at this point.

Reply:
If we do not wish to be specific in this way and infer the content then the format maybe should be JSON (ndjson?) with the understanding we will infer what it is though I like the idea of being explict:

* kubernetes event json
* if the parse or deserialization fails then its the senders fault because they sent us the wrong thing

Reply:
Check https://github.com/openshift/cluster-logging-operator/pull/2155 which has some changes in this regard. For hypershift we need at least k8s list format (events are in array .items[]) For general use we should have ndjson and maybe generalize to allow the user to specify an array field name for list formats that are similar to k8s but not identical. My minimum goal at this point is to release a preview that will handle the hypershift case and then add further minor features on z-stream.